### PR TITLE
feat: change Dialects.js to Dialect.ts

### LINKED ISSUE #640: Change `Dialects.js` to `Dialect.ts`
**Is your feature request related to a problem? Please describe.**
We want to start using TS in this project.

**Describe the solution you'd like**
Convert `Dialects.js` to `Dialect.ts`. The filename will now be singular. Therefore, all other files that used to rely on `Dialects` will need to be updated to import the newly renamed `Dialect` file.

### PR DESCRIPTION
**Pull request by Ebugo**
## Describe your changes

Change Dialects.js to Dialects.ts

## Issue ticket number and link

fix #640

### CODE REVIEW

File: src/controllers/utils/buildDocs.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ import omit from 'lodash/omit';
 import Versions from '../../shared/constants/Versions';
 import { wordSchema } from '../../models/Word';
 import { exampleSchema } from '../../models/Example';
-import Dialects from '../../shared/constants/Dialects';
+import Dialects from '../../shared/constants/Dialects.ts';
```
Comment (ijemmao) [inline src/controllers/utils/buildDocs.js:11]: we don't want to include extensions in our imports - this comment applies for the rest of your changes

**Review by ijemmao — CHANGES_REQUESTED** [review]
this is a great start! but please review the issue again, since it's requesting to change the name of the file from Dialects to Dialect - from plural to singular

**Review by ijemmao — CHANGES_REQUESTED** [review]

File: src/controllers/utils/buildDocs.js (hunk 1 above)
Comment (Ebugo) [reply src/controllers/utils/buildDocs.js]: Fixed @ijemmao

File: .eslintrc.js [hunk 2]
```diff
@@ -32,6 +31,16 @@ module.exports = {
...
+    'import/extensions': [
+      'error',
+      'ignorePackages',
+      {
+        js: 'never',
+        jsx: 'never',
+        ts: 'never',
+        tsx: 'never',
+      },
+    ],
```
Comment (ijemmao) [inline .eslintrc.js:43]: this change is technically out of scope for this PR - we would want to make a follow up issue to address this specifically - please revert the changes in this file

**Review by ijemmao — CHANGES_REQUESTED** [review]
real close! we just want to stay within scope of this issue - you can create a follow up issue to make an update to the eslintrc file

File: .eslintrc.js (hunk 2 above)
Comment (Ebugo) [reply .eslintrc.js]: Done

File: .eslintrc.js [hunk 3]
```diff
@@ -32,6 +31,7 @@ module.exports = {
     '@next/next/no-html-link-for-pages': ['off'],
     '@next/next/no-img-element': ['off'],
     'linebreak-style': ['error', os.platform() === 'win32' ? 'windows' : 'unix'],
+    'import/extensions': ['warn'],
```
Comment (ijemmao) [inline .eslintrc.js:34]: there shouldn't be any changes in this file

**Review by ijemmao — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 3 above)
Comment (Ebugo) [reply .eslintrc.js]: reverted to initial state

**Pull request merged**